Source/WebCore/platform/graphics/skia/GraphicsContextSkia.cpp

```diff
@@ -881,10 +881,13 @@ void GraphicsContextSkia::drawPattern(NativeImage& nativeImage, const FloatRect&
     if (!makeGLContextCurrentIfNeeded())
         return;
 
-    FloatRect rect(tileRect);
-    rect.moveBy(phase);
     SkMatrix phaseMatrix;
-    phaseMatrix.setTranslate(rect.x(), rect.y());
+    // Adjust phase based on tile offset and spacing.
+    double phaseOffsetX = phase.x() + tileRect.x() * patternTransform.a();
+    double phaseOffsetY = phase.y() + tileRect.y() * patternTransform.d();
```

Comment:
If understand the problem correctly, we are just missing the scale in the rect. So, just adding `rect.scale(patternTransform.a(), patternTransform.d());` before the moveBy would fix the problem, no? If that's the case, we can probably simplify this by using a FloatPoint instead of a rect, since now we only use the position, but not the size, something like this:

```
FloatPoint phaseOffset(tileRect.location());
phaseOffset.scale(patternTransform.a(), patternTransform.d());
phaseOffset.moveBy(phase);
SkMatrix phaseMatrix;
phaseMatrix.setTranslate(phaseOffset.x(), phaseOffset.y());
```